registry/handlers/manifests_test.go

```diff
@@ -0,0 +1,174 @@
+package handlers
+
+import (
+	"errors"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"fmt"
+
+	"time"
+
+	"github.com/docker/distribution/configuration"
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/reference"
+	"github.com/docker/distribution/registry/api/v2"
+	storagedriver "github.com/docker/distribution/registry/storage/driver"
+	"github.com/docker/distribution/registry/storage/driver/mockdriver"
+)
+
+var errUnexpected = errors.New("unexpected error")
+var testServer = createServer()
+
+func createServer() *httptest.Server {
```

Comment:
Thanks @stevvooe I will add the functionality of introducing storage error into TestDriver, and add the new test in api_test.go...plz let me if you suggest differently...